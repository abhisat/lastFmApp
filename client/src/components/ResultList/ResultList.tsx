import React, { SyntheticEvent } from "react";
import { ResultListItem } from "./ResultListItem/ResultListItem";
import { List, Container, Header } from "semantic-ui-react";
import Loader from "react-loader-spinner";
import { ResponseItem } from "../../types/ResponseTypes";
import "./ResultList.css";

export interface ResultListProps {
  title: string;
  feeds: Array<ResponseItem>;
  isLoading: boolean;
  handleFeedClick?(value: string, e: SyntheticEvent): void;
}

const ResultList: React.FunctionComponent<ResultListProps> = (
  props: ResultListProps
) => {
  const { title, feeds, isLoading, handleFeedClick } = props;

  const feed = feeds ? (
    feeds.map((feedItem: ResponseItem, id: number) => {
      return (
        <List.Item key={id + feedItem.name + feedItem.url}>
          <ResultListItem
            name={feedItem.name}
            playcount={feedItem.playcount}
            listeners={feedItem.listeners}
            url={feedItem.url}
            image={feedItem.image[0]["#text"]}
            isLoading={isLoading}
            handleClick={handleFeedClick}
          />
        </List.Item>
      );
    })
  ) : (
    <Loader type='TailSpin' color={"#b90000"} height={100} width={100} />
  );
  const headerContent = feeds
    ? feeds.length !== 0
      ? `... Showing ${title}`
      : `Sorry no content was found.`
    : null;
  return (
    <React.Fragment>
      {isLoading ? null : (
        <Header className={"resultHeader"} as='h2'>
          {headerContent}
        </Header>
      )}
      <Container className={"feedContainer"}>
        <List divided className={"list"}>
          {feed}
        </List>
      </Container>
    </React.Fragment>
  );
};

export { ResultList };
