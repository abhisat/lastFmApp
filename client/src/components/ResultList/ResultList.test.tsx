import React from "react";
import { ResultList, ResultListProps } from "./ResultList";
import EnzymeAdapter from "enzyme-adapter-react-16";
import Enzyme, { shallow } from "enzyme";
import toJson from "enzyme-to-json";

Enzyme.configure({
  adapter: new EnzymeAdapter()
});

const handleFeedClick: jest.Mock = jest.fn();
const handlePaginationClick: jest.Mock = jest.fn();

const props: ResultListProps = {
  title: "Uploads from everyone",
  feeds: [
    {
      name: "Arctic Monkeys",
      playcount: "349443530",
      listeners: "3618819",
      mbid: "ada7a83c-e3e1-40f1-93f9-3e73dbc9298a",
      url: "https://www.last.fm/music/Arctic+Monkeys",
      streamable: "0",
      image: [
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "small"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "medium"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/174s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "large"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "extralarge"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "mega"
        }
      ]
    },
    {
      name: "Tyler, the Creator",
      playcount: "71334421",
      listeners: "934630",
      mbid: "f6beac20-5dfe-4d1f-ae02-0b0a740aafd6",
      url: "https://www.last.fm/music/Tyler,+the+Creator",
      streamable: "0",
      image: [
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "small"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/64s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "medium"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/174s/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "large"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "extralarge"
        },
        {
          "#text":
            "https://lastfm.freetls.fastly.net/i/u/300x300/2a96cbd8b46e442fc41c2b86b821562f.png",
          size: "mega"
        }
      ]
    }
  ],
  isLoading: false,
  handleFeedClick: handleFeedClick
};

const mountApp = () => {
  const wrapper: Enzyme.ShallowWrapper = shallow(<ResultList {...props} />);
  wrapper.update();
  return wrapper;
};

describe("<ResultList/>", () => {
  let App: Enzyme.ShallowWrapper;

  beforeEach(() => {
    App = mountApp();
  });

  afterEach(() => {});

  it("renders without crashing", () => {
    mountApp();
  });

  it("Matches the snapshot", () => {
    expect(toJson(App)).toMatchSnapshot();
  });
});
