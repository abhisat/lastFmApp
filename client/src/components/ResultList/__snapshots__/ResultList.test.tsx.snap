// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultList/> Matches the snapshot 1`] = `
<Fragment>
  <Header
    as="h2"
    className="resultHeader"
  >
    ... Showing Uploads from everyone
  </Header>
  <Container
    className="feedContainer"
  >
    <List
      className="list"
      divided={true}
    >
      <ListItem
        key="0Arctic Monkeyshttps://www.last.fm/music/Arctic+Monkeys"
      >
        <ResultListItem
          handleClick={[MockFunction]}
          image="https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png"
          isLoading={false}
          listeners="3618819"
          name="Arctic Monkeys"
          playcount="349443530"
          url="https://www.last.fm/music/Arctic+Monkeys"
        />
      </ListItem>
      <ListItem
        key="1Tyler, the Creatorhttps://www.last.fm/music/Tyler,+the+Creator"
      >
        <ResultListItem
          handleClick={[MockFunction]}
          image="https://lastfm.freetls.fastly.net/i/u/34s/2a96cbd8b46e442fc41c2b86b821562f.png"
          isLoading={false}
          listeners="934630"
          name="Tyler, the Creator"
          playcount="71334421"
          url="https://www.last.fm/music/Tyler,+the+Creator"
        />
      </ListItem>
    </List>
  </Container>
</Fragment>
`;
